src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets

```diff
@@ -173,7 +209,9 @@ The .NET Foundation licenses this file to you under the MIT license.
       <LinkerArg Include="@(NativeFramework->'-framework %(Identity)')" Condition="'$(_IsApplePlatform)' == 'true'" />
       <LinkerArg Include="-Wl,--eh-frame-hdr" Condition="'$(_IsApplePlatform)' != 'true'" />
     </ItemGroup>
+  </Target>
 
+  <Target Name="VerifyOSSpecificTooling">
```

Comment:
Why do we need to split this? More places than just tests.singlefile.targets refer to the existing target (you can search github). I would rather not break them if I can help it.

Reply:
I made the split for two reasons:
1. To avoid moving the code that probes OS-specific tooling to the beginning of the `SetupOSSpecificProps`
    - This is required because we need `xcrun` to be verified before populating all the Apple-specific properties introduced in this PR
2. To separate the logic in `SetupOSSpecificProps` as it is: 
    - Setting up various properties and items, and 
    - Verifying the existence of OS-specific tools that some of these properties and items depend on

I could remove the newly added target `VerifyOSSpecificTooling` and reorganize `SetupOSSpecificProps` to make all this work together, but I think it is cleaner to have a separate target. What do you think?

Reply:
> To avoid moving the code that probes OS-specific tooling to the beginning

I might be missing something, but can we just take lines 284-294 and paste them before line 60? The only "problematic" property would be `_CommandProbe` because it's not defined there, but we can just replace `$(_CommandProbe)` with `command -v`. We already don't use `$(_CommandProbe)` with Apple specific stuff because cross compilation from non-Apple to Apple platform is an academic scenario.

> I think it is cleaner to have a separate target

`VerifyOSSpecificTooling` would still be doing stuff like compiling System.Globalization.Native for StaticICULinking and `SetupOSSpecificProps` would still be checking for a valid SDK with xcrun. It doesn't feel like it moves the needle. Many 3rd party repos depend on this target and this is not a good reason to break them.

Reply:
I see, thank you for the suggestion and clarification! 
I will remove the `VerifyOSSpecificTooling` and adjust the existing target accordingly.